src/go/rpk/pkg/vcloud/auth0.go

```diff
@@ -0,0 +1,134 @@
+	auth0Domain string
+}
+
+const (
+	// CLI clientId
+	// TODO(av) replace with production clientId
+	clientId = "yMMbfD6xdKXW9DmIqAZDzTBPHgfI5MyX"
+	// TODO(av) should be configurable
+	defaultAuth0Domain = "vectorized-dev.us.auth0.com"
+
+	// url path to retrieve device code
+	deviceCodePath = "oauth/device/code"
+	// url path to retrieve token
+	tokenPath = "oauth/token"
+)
+
+func NewDefaultAuth0Client() AuthClient {
+	return &Auth0Client{
+		clientId:    clientId,
+		auth0Domain: defaultAuth0Domain,
+	}
+}
+
+// getToken queries auth0 token endpoint waiting for token to be issued
+// https://auth0.com/docs/flows/call-your-api-using-the-device-authorization-flow#request-tokens
+func (ac *Auth0Client) GetToken(deviceCode string) (*TokenResponse, error) {
+	requestBody := []byte(fmt.Sprintf("grant_type=urn:ietf:params:oauth:grant-type:device_code&device_code=%s&client_id=%s", deviceCode, clientId))
+	deviceCodeUrl := fmt.Sprintf("https://%s/%s", defaultAuth0Domain, tokenPath)
+	req, err := http.NewRequest("POST", deviceCodeUrl, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return nil, fmt.Errorf("error creating token request. %w", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error calling token api: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading token step body: %w", err)
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("unexpected status code. 200 != %d. Body: %s", resp.StatusCode, body)
+	}
+	var response TokenResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling token response: %w", err)
+	}
+	return &response, err
+}
+
+func (ac *Auth0Client) GetDeviceCode(
+	audience string,
+) (*DeviceCodeResponse, error) {
+	requestBody := []byte(fmt.Sprintf("client_id=%s&audience=%s", clientId, yakAudience))
+	deviceCodeUrl := fmt.Sprintf("https://%s/%s", defaultAuth0Domain, deviceCodePath)
+	req, err := http.NewRequest("POST", deviceCodeUrl, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return nil, fmt.Errorf("error creating device code request. %w", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error calling device code api: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading device code step body: %w", err)
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("unexpected status code. 200 != %d. Body: %s", resp.StatusCode, body)
+	}
+	var response DeviceCodeResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling the device code response: %w", err)
+	}
+	return &response, err
```

Comment:
err is nil here, so I think it's better to just be explicit and do `return &response, nil`